kubernetes/client/api/custom_objects_api.py

```diff
@@ -2573,8 +2575,8 @@ def patch_cluster_custom_object_scale_with_http_info(self, group, version, plura
             ['application/json', 'application/yaml', 'application/vnd.kubernetes.protobuf'])  # noqa: E501
 
         # HTTP header `Content-Type`
-        header_params['Content-Type'] = self.api_client.select_header_content_type(  # noqa: E501
-            ['application/merge-patch+json'])  # noqa: E501
+        header_params['Content-Type'] = local_var_params.get('content_type', self.api_client.\
+            select_header_content_type(['application/merge-patch+json'], 'PATCH', body_params))
```

Comment:
want to double check, this PR would be backward compatible for both core native types, and custom resources, right?

Reply:
It's tricky question, in general it's backward compatible, but I can imagine that some calls can be interpreted differently by select_header_content_type - the current version usually uses the first patch method but this version tries to guess the content-type and results may be different... To be honest I don't think it's a real problem.

The positive aspect is that we will be able to stop removing the first element from the list as in this change https://github.com/roycaihw/client-python/commit/61f1fbe49af3be34cebe8781efa436ae8b84e2e2 and select_header_content_type should do the job.

Reply:
sounds good to me, @roycaihw do you have any concern?

Reply:
+1 to it's not a real problem (at least for this repo, given how we use `select_header_content_type` today).

Today native types can only use `application/json-patch+json` or `application/strategic-merge-patch+json`. With this change, existing applications will still get the same result. New applications can use the exposed `content_type` to choose using `application/merge-patch+json`.

Custom resources today can only use `application/merge-patch+json` and will get the same result. With this change, applications will be able to pass in `application/json-patch+json` body.